modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/item/selector/ObjectEntryItemSelectorViewDescriptor.java

```diff
@@ -139,10 +139,21 @@ public SearchContainer<ObjectEntry> getSearchContainer()
 							_portletRequest, "objectRelationshipId")));
 			}
 			else {
+				Group scopeGroup = _themeDisplay.getScopeGroup();
+
+				Page<com.liferay.object.rest.dto.v1_0.ObjectEntry> page =
+					_objectEntryManager.getObjectEntries(
+						_themeDisplay.getCompanyId(), _objectDefinition,
+						scopeGroup.getGroupKey(), null, _getDTOConverterContext(),
+						StringPool.BLANK, Pagination.of(searchContainer.getCur(), searchContainer.getDelta()), ParamUtil.getString(_portletRequest, "keywords"),
+						null);
+
 				searchContainer.setResultsAndTotal(
-					_getObjectEntries(
-						searchContainer.getCur(), searchContainer.getDelta(),
-						ParamUtil.getString(_portletRequest, "keywords")));
+					() -> TransformUtil.transform(
+						page.getItems(),
+						objectEntry -> ObjectEntryUtil.toObjectEntry(
+							_objectDefinition.getObjectDefinitionId(), objectEntry)),
+					Math.toIntExact(page.getTotalCount()));
```

Comment:
Hey @thallescezar , can you use `GetterUtil.getInteger(page.getTotalCount());`? Like this [example](https://github.com/liferay/liferay-portal/blob/cc7034192fede4a5dee2cad027c7a650e87456fe/modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectFieldResourceTest.java#L135). Thanks!